### PR TITLE
roles(associated roles+attributes): address UX feedback

### PR DESCRIPTION
**Pull request by jenny-s51**
## Motivation

Towards #416 

## Verification Steps

1. Go to Realm Roles section.
2. Click on a role. 
3. Add a few associated roles.
4. Verify that the roles are alphabetized. 
5. Search for a particular associated role in the search input field. 
6. Click the search button and verify that filtering works as expected.
7. Go to Role Attributes. 
8. Verify it matches the new design: https://marvelapp.com/prototype/6j3ecj2/screen/77563386, and adding/saving attributes works as expected

## Checklist:

- [x] Code has been tested locally by PR requester
- [x] User-visible strings are using the react-i18next framework (useTranslation)
- [x] Help has been implemented
- [x] axe report has been run and resulting a11y issues have been resolved
- [x] Unit tests have been created/updated
- [x] Formatting has been performed via prettier/eslint
- [x] Type checking has been performed via 'yarn check-types'

### CODE REVIEW

File: src/realm-roles/AssociatedRolesTab.tsx [hunk 1]
```diff
@@ -170,6 +192,8 @@ export const AssociatedRolesTab = ({
     },
   });
 
+  console.log(selectedRows);
```
Comment (ssilvert) [inline src/realm-roles/AssociatedRolesTab.tsx:195]: Remove.

**Review by ssilvert — CHANGES_REQUESTED** [review]
Looks good except for one stray console log.  Need someone on UXD team to review before we merge though.

**Comment by yih-wang** [comment]
The search of the Add associated roles modal doesn't work. 
image
image

**Comment by jenny-s51** [comment]
@yih-wang thank you for your feedback! I've addressed the `revert` in this PR , the pagination/search in #460, and opened a separate issue for the implementation of disabled kebabs/checkboxes. Please kindly review again when you have a moment :)

**Review by mfrances17 — CHANGES_REQUESTED** [review]
When testing the Attributes tab, the Key and Value fields only show up for 1 of my 5 defined roles - the design says they should be visible by default. So obviously i’m only able to add, save, etc on this one role. Is this working as expected - is this prevented for those other 4 roles because of permission(s) or the type of defined role? this by design or is it an issue?

**Comment by jenny-s51** [comment]
@mfrances17 thank you for your comment; nice catch! There was no initial key/value field in the attributes tab as soon as a role was created. I just fixed this - let me know if you're able to see the fields now 🙂

**Review by mfrances17 — APPROVED** [review]
Ok, great... your latest push fixes the issue I was experiencing with no default key/value empty fields. LGTM.

**Pull request merged**